tmt/uuid.py

```diff
@@ -0,0 +1,109 @@
+from typing import Optional
+from uuid import uuid4
+
+import fmf
+
+from .utils import log
+
+ID_FMF_ITEM = "id"
+
+
+class IDError(Exception):
+    pass
+
+
+class IDLeafError(IDError):
+    pass
+
+
+def locate_key(node: fmf.Tree, key: str) -> Optional[fmf.Tree]:
+    """
+    Return fmf location of key attribute inside tree
+    """
+    identifier = node.data.get(key)
+    if identifier is None:
+        raise IDError("ID not defined")
```

Comment:
should be `return None`, like this it prevents any automatic generation (e.g. during export) because it will always fail here

Reply:
From my perspective returning `None` or exception is same, and exception avoid to misinterpret return value if none and in your code you have to have try except block to handle it in export, so that your code will be something like:
```
try:
    id = self.id
except IDError:
    id = tmt.uuid,add_uuid_if_not_defined(self.node)
...
```

Reply:
Yes, basically both are possible. I'd lean towards `None` as proposed by @KwisatzHaderach though as this seems to be more natural: The function searches for the node and returns it if found and `None` if nothing found.

Reply:
Thanks for changing this to `None`.